### ISSUE TITLE
[Feature Request]: Use separate VAE

### ISSUE THREAD

**Issue by catboxanon** [Bing-su/adetailer#307]
### Is your feature request related to a problem? Please describe.

When inpainting with anime models, one problem that exists is that, the VAE that should ideally be used with those (`animevae.pt`) causes the entire output to be desaturated. This can be easily observed by using this VAE in the img2img tab, inputting an image, and generating with denoising set to `0` (in other words, encoding the image and decoding the resulting latent). The [kl-f8](https://github.com/CompVis/latent-diffusion#pretrained-autoencoding-models) based VAEs (such as [the one StabilityAI finetuned on faces](https://huggingface.co/stabilityai/sd-vae-ft-mse-original)) do not have this issue, and in all scenarios are the ones that should be used for inpainting.

However, in txt2img, the desaturation issue with `animevae.pt` is not important, as we always start from no image, and the final result can be alleviated by color correction or postprocessing extensions like [sd-webui-color-enhance](https://git.mmaker.moe/mmaker/sd-webui-color-enhance.git). There's also been plenty of examples shown that using this VAE in txt2img handles aspects like contrast and details like eyes better since it was (assumed to be) finetuned on anime-styled artwork.

**tl;dr**: due to desaturation issues with a specific VAE in img2img, but better results in txt2img, there should be an ability to use a separate VAE.

### Describe the solution you'd like

Add an additional option within adetailer to specify a separate VAE used during adetailer processing, which would then reload the VAE specified within the webui upon processing completion.

### Describe alternatives you've considered

The only alternative I see is to not use adetailer in txt2img, and instead to use it within the img2img tab after generating an image in txt2img, and after selecting a different VAE within the webui. This sort of defeats the purpose of adetailer though.

### Additional context

Info from here may be useful reference.
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/6074#issuecomment-1668256099

~~I am also curious to look into this specific VAE a bit more, because due to it's origin, and it's origin software supporting img2img, I would actually be surprised it was put into production like this. That's a bit more unrelated to this issue though and I still believe having this option available is useful.~~ I did a quick check and using the origin software indeed has the same issue, meaning it's a fundemental issue with `animevae.pt`.
